app/services/file_attachment_blob_service.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class FileAttachmentBlobService
+  attr_reader :file, :revision, :replacement
+
+  def initialize(file:, revision:, replacement: nil)
+    @file = file
+    @revision = revision
+    @replacement = replacement
+  end
+
+  def blob_id
+    blob.id
+  end
+
+  def filename
```

Comment:
It's likely worth memoing this you could wrap it in a begin end

@filename ||= begin
  # code
end

Reply:
Done 👍